riscv/insns/froundnx_d.h

```diff
@@ -0,0 +1,17 @@
+require_extension('D');
+require_extension(EXT_ZFA);
+require_fp;
+{
+  bool isNaN = isNaNF64UI(FRS1_D.v);
+  bool isZero = fracF64UI(FRS1_D.v) == 0 && expF64UI(FRS1_D.v) == 0;
+  if (isNaN || isZero) {
+    /* set invalid operation flag for signalling NaN */
+    if (f64_isSignalingNaN(FRS1_D))
+      STATE.fflags->write(STATE.fflags->read() | softfloat_flag_invalid);
+    /* NaN or positive/negative zero are copied unmodified */
+    WRITE_FRD_D(FRS1_D);
```

Comment:
NaNs must be canonicalized.

Reply:
Similarly, why doesn't `f64_roundToInt(FRS1_D, RM, false)` fully suffice for the entirety of this instruction?